### ISSUE TITLE
Document version matrix in provider getting started documentation (gitbook)

### ISSUE THREAD

**Issue by chuckha** [kubernetes-sigs/cluster-api#807]
/kind documentation
/milestone Next
/priority important-longterm

**Describe the solution you'd like**
We should add the version matrix to the getting started implementing a provider section of the gitbook. The markdown that we are using looks like this:

```
||Cluster API 0.1|Cluster API 0.2|
|-|-|-|
|XYZ Provider 0.1|✓||
|XYZ Provider 0.2||✓|

||Kubernetes 1.11|Kubernetes 1.12|Kubernetes 1.13|Kubernetes 1.14|
|-|-|-|-|-|
|XYZ Provider 0.1|✓|✓|✓|
|XYZ Provider 0.2||✓|✓|✓|
```

Which yields tables like this:

||Cluster API 0.1|Cluster API 0.2|
|-|-|-|
|XYZ Provider 0.1|✓||
|XYZ Provider 0.2||✓|

||Kubernetes 1.11|Kubernetes 1.12|Kubernetes 1.13|Kubernetes 1.14|
|-|-|-|-|-|
|XYZ Provider 0.1|✓|✓|✓|
|XYZ Provider 0.2||✓|✓|✓|

**Comment by ncdc** [comment]
Closing in favor of updating the provider docs once clusterctl v2 is ready. Tracking issue is #1729

/close
